assets/js/base/hooks/checkout/use-checkout-address.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import defaultAddressFields from '@woocommerce/base-components/cart-checkout/address-form/default-address-fields';
+import { useState, useCallback, useEffect } from '@wordpress/element';
+import {
+	useShippingDataContext,
+	useBillingDataContext,
+	useCheckoutContext,
+} from '@woocommerce/base-context';
+
+/**
+ * Compare two addresses and see if they are the same.
+ *
+ * @param {Object} address1 First address.
+ * @param {Object} address2 Second address.
+ */
+const isSameAddress = ( address1, address2 ) => {
+	const diff = Object.keys( defaultAddressFields ).filter( ( field ) => {
+		return address1[ field ] !== address2[ field ];
+	} );
+	return diff.length === 0;
+};
```

Comment:
this could be simplified with `Array.every`

```suggestion
const isSameAddress = ( address1, address2 ) => {
	return Object.keys( defaultAddressFields ).every( ( field ) => address1[ field ] === address2[ field ] );
};
```